### PR TITLE
NEW Add ElementalAreasPageExtension

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-elemental/issues/1215

### CODE REVIEW

File: code/elemental/ElementalAreasPageExtension.php
Comment (ChloeSartorelli) [inline code/elemental/ElementalAreasPageExtension.php]: Why not `extends ElementalPageExtension`?

It would satisfy me if we just say "this doesn't have to be for a page, it can be for any dataobject" and then just make this change:

```suggestion
 * This is used to test multiple elemental areas on a single data object
 */
class MultiElementalAreasExtension extends ElementalAreasExtension
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: code/elemental/ElementalAreasPageExtension.php
Comment (emteknetnz) [reply code/elemental/ElementalAreasPageExtension.php]: Rather confusingly, `ElementalPageExtension extends ElementalAreasExtension`

`ElementalAreasExtension` is what enables you to have on or more `ElementalArea`'s on `DataObject` and things get saved correctly

`ElementalPageExtension` is an essentially an implementation of that and it adds in single ElementalArea relation

However for the behat test I'm doing it's explicitly testing 2x ElementalArea's (testing more complexity that way) so I think it's much cleaner to extend the base `ElementalAreasExtension`

I'll update the naming

**Review by GuySartorelli — APPROVED** [review]
LGTM, will merge when other PRs are approved

**Comment by emteknetnz** [comment]
Have made a change after it was approved

**Review by GuySartorelli — APPROVED** [review]
Could have just reused existing modeladmin and object, and applied the extension in behat... but this will do.

**Pull request merged**
